go-controller/pkg/ovn/gateway_init.go

```diff
@@ -386,6 +386,16 @@ func gatewayInit(nodeName string, clusterIPSubnet []*net.IPNet, hostSubnets []*n
 					entry.String(), gatewayRouter, err)
 			}
 		}
+	} else {
+		// ensure we do not have any leftover SNAT entries after an upgrade
```

Comment:
probably should be adding a utest to capture this.

Reply:
done